src/GameSync.ViewModels/Common/RelayCommand.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Windows.Input;
+
+namespace GameSync.ViewModels.Common;
+
+public class RelayCommand : ICommand
+{
+    Action _TargetExecuteMethod;
+    Func<bool> _TargetCanExecuteMethod;
+
+    public RelayCommand(Action executeMethod)
+    {
+        _TargetExecuteMethod = executeMethod;
+    }
+
+    public RelayCommand(Action executeMethod, Func<bool> canExecuteMethod)
+    {
+        _TargetExecuteMethod = executeMethod;
+        _TargetCanExecuteMethod = canExecuteMethod;
+    }
+
+    public void RaiseCanExecuteChanged()
+    {
+        CanExecuteChanged(this, EventArgs.Empty);
+    }
+    #region ICommand Members
+
+    bool ICommand.CanExecute(object parameter)
+    {
+        if (_TargetCanExecuteMethod != null)
+        {
+            return _TargetCanExecuteMethod();
+        }
+        if (_TargetExecuteMethod != null)
```

Comment:
How about `return _TargetExecuteMethod != null;`? Probably same goes in other RelayCommand classes

Reply:
I admit that I `ctrl + C` and `ctrl + V` this code from Microsoft's blog post about MVVM :)

Reply:
Done